### PR TITLE
Add map_to_object/2

### PR DESCRIPTION
**Pull request by vans163**
This one should be final.

### CODE REVIEW

File: src/javascript/lib/core/functions.js [hunk 1]
```diff
@@ -92,12 +94,26 @@ function build_namespace(ns, ns_string) {
...
 
-  for (const [key, value] of map.entries()) {
+  var type_keys = proplists.get_value(Symbol("keys"), options);
+  var symbols = proplists.get_value(Symbol("symbols"), options);
+
+  for (var [key, value] of map.entries()) {
+    if (type_keys == Symbol("string") && typeof key == 'number') {
+      key = key.toString();
+    } else if (
+        (type_keys == Symbol("string") || symbols != Symbol("undefined"))
```
[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/lib/core/functions.js]: No!

File: src/javascript/tests/core/functions.spec.js [hunk 2]
```diff
@@ -24,3 +24,16 @@ test('split_at', t => {
...
+  const map = new Map([
+    [Symbol.for('key'), "value"],
+    [Symbol.for('anotherKey'), "value2"],
+  ]);
+
+  const options = [new Core.Tuple(Symbol.for('keys'),Symbol.for('strings'))];
+
+  const result = Functions.map_to_object(map, options);
+
+  t.deepEqual(result, {key: "value", anotherKey: "value"});
```
[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: makes no sense, why?

[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: No!

[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: No No!

[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: now its required after? lol

File: src/javascript/tests/core/functions.spec.js [hunk 3]
```diff
@@ -24,3 +24,16 @@ test('split_at', t => {
...
   t.deepEqual(Functions.split_at('😀abélkm', 4).values, ['😀abé', 'lkm']);
 });
+
+test('map_to_object/2', t => {
+  const map = new Map([
+    [Symbol.for('key'), "value"],
+    [Symbol.for('anotherKey'), "value2"],
+  ]);
+
+  const options = [new Core.Tuple(Symbol.for('keys'),Symbol.for('strings'))];
```
[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: okay this one may be right,

File: src/javascript/tests/core/functions.spec.js [hunk 4]
```diff
@@ -24,3 +24,16 @@ test('split_at', t => {
   t.deepEqual(Functions.split_at('abc', -1000).values, ['', 'abc']);
   t.deepEqual(Functions.split_at('😀abélkm', 4).values, ['😀abé', 'lkm']);
 });
+
+test('map_to_object/2', t => {
+  const map = new Map([
+    [Symbol.for('key'), "value"],
+    [Symbol.for('anotherKey'), "value2"],
```
[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: No No No!

File: src/javascript/tests/core/functions.spec.js [hunk 5]
```diff
@@ -24,3 +24,16 @@ test('split_at', t => {
   t.deepEqual(Functions.split_at('abc', -1000).values, ['', 'abc']);
   t.deepEqual(Functions.split_at('😀abélkm', 4).values, ['😀abé', 'lkm']);
 });
+
+test('map_to_object/2', t => {
+  const map = new Map([
+    [Symbol.for('key'), "value"],
```
[the remark this answers is not in the document]
Comment (vans163) [reply src/javascript/tests/core/functions.spec.js]: ...

**Comment by bryanjos** [comment]
Looks good! I guess you see I turned on ebert for code reviews. Still thinking about changing settings.  Some of them are valid here, but I can still merge this and fix them later

**Comment by vans163** [comment]
The bot seems very aggressive to pick up too many small things, or weird things like 3 space indents / single quote strings.

Also its hard to see the full changed diff, its interweaved with the bot comments.

**Pull request merged**